codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -247,23 +245,27 @@ static State activeState(int streamId, State initialState, boolean isLocal, bool
         }
     }
 
-    void notifyHalfClosed(Http2Stream stream) {
-        for (int i = 0; i < listeners.size(); i++) {
-            try {
+    final void notifyHalfClosed(Http2Stream stream) throws Http2Exception {
+        try {
+            for (int i = 0; i < listeners.size(); i++) {
```

Comment:
not sure if it matters but maybe we want to call size() only one time and store it in temp variable ? Same for all other for loops.